weld-rendering/app/src/main/java/styler/weld/rendering/viewholder/WidgetFactory.kt

```diff
@@ -1,31 +1,60 @@
 package styler.weld.rendering.viewholder
 
+import android.view.LayoutInflater
 import android.view.ViewGroup
+import styler.weld.rendering.R
 
 class WidgetFactory {
 
     var factories: MutableList<FactoryMethod> = mutableListOf();
 
-    fun register(type: String, factory: (parent: ViewGroup) -> BaseViewHolder) {
-        factories.add(FactoryMethod(type, factory));
+    fun register(type: String) {
+        factories.add(FactoryMethod(type))
     }
 
-    fun createViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
+    fun createViewHolder(viewType: Int, parent: ViewGroup): BaseViewHolder {
+        if (viewType == INVALID) {
+            // TODO: INVALID case should be refactored later
+            val view = LayoutInflater.from(parent.context)
+                .inflate(R.layout.empty_view_row_item, parent, false)
+            return EmptyViewHolder(view)
+        }
         return factories[viewType].createViewHolder(parent)
     }
 
     fun getViewTypeFor(type: String): Int {
-        // should default to a -1 or something to signify an unsupported type
-        var factory = factories.findLast { it.type == type }
-        if (factory == null) return fallback
-        return factories.indexOf(factory)
+        val factory = factories.findLast { it.type == type }
+        return if (factory == null) {
+            INVALID
+        } else {
+            factories.indexOf(factory)
+        }
     }
 
     companion object {
-        const fallback: int = -1;
+        const val INVALID = -1
     }
 }
 
-class FactoryMethod(type: String, factory: (parent: ViewGroup) -> BaseViewHolder) {
+class FactoryMethod(val type: String) {
```

Comment:
nope :) this is what we want to avoid. use inheritance or lambdas to register these. avoid switch cases

https://blog.cleancoder.com/uncle-bob/2021/03/06/ifElseSwitch.html

https://www.linkedin.com/pulse/why-switch-statements-bad-th%C3%A9o-farnole-/?trk=public_profile_article_view

Reply:
Thanks for the article, changed to use lambda to register in `MainActivity`